SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -595,12 +595,24 @@ void USpatialReceiver::ApplyComponentData(Worker_EntityId EntityId, Worker_Compo
 
 	ESchemaComponentType ComponentType = TypebindingManager->FindCategoryByComponentId(Data.component_id);
 
+
 	if (ComponentType == SCHEMA_Data || ComponentType == SCHEMA_OwnerOnly)
 	{
+		if (UActorComponent* ActorComp = Cast<UActorComponent>(TargetObject))
+		{
+			Schema_Object* ComponentObject = Schema_GetComponentDataFields(Data.schema_type);
+			bool bReplicates = !!Schema_IndexBool(ComponentObject, SpatialConstants::ACTOR_COMPONENT_REPLICATES_ID, 0);
```

Comment:
I think we should put a `check` into schema generator to ensure `ACTOR_COMPONENT_REPLICATES_ID` indeed corresponds to `bReplicates`. I know it does now, but this will make it a bit less fragile in terms of future changes.